Services/CO.CDP.DataSharing.WebApi/AutoMapper/DataSharingProfile.cs

```diff
@@ -23,8 +28,98 @@ public DataSharingProfile()
           .ForMember(m => m.SubmittedAt, o => o.MapFrom(m => m.SubmittedAt))
           .ForMember(m => m.ShareCode, o => o.MapFrom(m => m.ShareCode));
+
+        CreateMap<Persistence.SharedConsent, SupplierInformation>()
+          .ForMember(m => m.Id, o => o.MapFrom(m => m.Organisation.Guid))
+          .ForMember(m => m.Name, o => o.MapFrom(m => m.Organisation.Name))
+          .ForMember(m => m.AdditionalParties, o => o.MapFrom(m => Enumerable.Empty<OrganisationReference>()))
+          .ForMember(m => m.Identifier, o => o.MapFrom(m => m.Organisation.Identifiers.FirstOrDefault(x => x.Primary)))
+          .ForMember(m => m.AdditionalIdentifiers, o => o.MapFrom(m => m.Organisation.Identifiers.Where(x => !x.Primary).ToList()))
+          .ForMember(m => m.Address, o => o.MapFrom(m => m.Organisation.Addresses.FirstOrDefault(x => x.Type == OrganisationInformation.AddressType.Registered)))
+          .ForMember(m => m.ContactPoint, o => o.MapFrom(m => m.Organisation.ContactPoints.FirstOrDefault()))
+          .ForMember(m => m.Roles, o => o.MapFrom(m => m.Organisation.Roles))
+          .ForMember(m => m.Details, o => o.MapFrom(m => new Details()))
+          .ForMember(m => m.SupplierInformationData, o => o.MapFrom(m => m));
+
+        CreateMap<OrganisationInformation.Persistence.Organisation.Identifier, OrganisationInformation.Identifier>()
+          .ForMember(m => m.Scheme, o => o.MapFrom(m => m.Scheme))
+          .ForMember(m => m.Id, o => o.MapFrom(m => m.IdentifierId))
+          .ForMember(m => m.LegalName, o => o.MapFrom(m => m.LegalName))
+          .ForMember(m => m.Uri, o => o.MapFrom(m => !string.IsNullOrWhiteSpace(m.Uri) ? new Uri(m.Uri) : default));
+
+        CreateMap<OrganisationInformation.Persistence.Organisation.OrganisationAddress, OrganisationInformation.Address>()
+          .ForMember(m => m.StreetAddress, o => o.MapFrom(m => m.Address.StreetAddress))
+          .ForMember(m => m.Locality, o => o.MapFrom(m => m.Address.Locality))
+          .ForMember(m => m.Region, o => o.MapFrom(m => m.Address.Region))
+          .ForMember(m => m.PostalCode, o => o.MapFrom(m => m.Address.PostalCode))
+          .ForMember(m => m.CountryName, o => o.MapFrom(m => m.Address.CountryName))
+          .ForMember(m => m.Country, o => o.MapFrom(m => m.Address.Country))
+          .ForMember(m => m.Type, o => o.MapFrom(m => m.Type));
+
+        CreateMap<OrganisationInformation.Persistence.Organisation.ContactPoint, OrganisationInformation.ContactPoint>()
+          .ForMember(m => m.Name, o => o.MapFrom(m => m.Name))
+          .ForMember(m => m.Email, o => o.MapFrom(m => m.Email))
+          .ForMember(m => m.Telephone, o => o.MapFrom(m => m.Telephone))
+          .ForMember(m => m.Url, o => o.MapFrom(m => !string.IsNullOrWhiteSpace(m.Url) ? new Uri(m.Url) : default));
+
+        CreateMap<OrganisationInformation.PartyRole, OrganisationInformation.PartyRole>();
```

Comment:
Is this necessary?

Reply:
@jakzal and @maciej-goaco 
I can see lots of mapping done for Supplier Information , the supplier information holds the following data as well but I cannot see that is mapped. I am not sure should this be mapped as well?  There is no description in the PR :) 


  public List<Qualification> Qualifications { get; set; } = [];
  public List<TradeAssurance> TradeAssurances { get; set; } = [];
  public LegalForm? LegalForm { get; set; }

Reply:
This will be part of the `detail` of supplier information. Good spot 👍